pkg/virt-api/webhooks/mutating-webhook/mutators/vmi-mutator.go

```diff
@@ -132,17 +132,33 @@ func (mutator *VMIsMutator) setDefaultMachineType(vmi *v1.VirtualMachineInstance
 }
 
 func (mutator *VMIsMutator) setDefaultResourceRequests(vmi *v1.VirtualMachineInstance) {
-	if _, exists := vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory]; !exists {
-		if vmi.Spec.Domain.Resources.Requests == nil {
-			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{}
+	resources := &vmi.Spec.Domain.Resources
+
+	if !resources.Limits.Cpu().IsZero() && resources.Requests.Cpu().IsZero() {
+		if resources.Requests == nil {
+			resources.Requests = k8sv1.ResourceList{}
+		}
+		resources.Requests[k8sv1.ResourceCPU] = resources.Limits[k8sv1.ResourceCPU]
```

Comment:
I wonder what is the expected result of:
cpu-request = 0
guest-cpu = 4 cpus
cpu-limit = 8000m
I read this as "I want my VM to have 4 vcpus but should never consume more than 8 cpus", right?
This code would set cpu-request=8000m which means that for scheduling purposes we require 8 cpus for this VM although this is more than what it really needs. I think that unlike ordinary pods, we should take into account the guest cpus here

Reply:
I want to be *Pod* compatible. In order to get req: 4, limit: 8 and guest:4 I would expect that you set all three values.